internal/generator/vector/pipelines.go

```diff
@@ -42,8 +70,24 @@ if .log_type == "application" {
 		}
 		inputs := []string{}
 		for _, inputName := range p.InputRefs {
-			if _, ok := userDefined[inputName]; ok {
-				inputs = append(inputs, fmt.Sprintf(UserDefinedInput, inputName))
+			if input, ok := userDefined[inputName]; ok {
+				if input.Application.ContainerLimit != nil {
+					if input.Application.ContainerLimit.Policy == logging.DropPolicy {
+						el = append(el, AddThrottle(input)...)
+						inputs = append(inputs, fmt.Sprintf(UserDefinedSourceThrottle, input.Name))
+					} else {
+						log.V(1).Info("Not adding source in pipeline due to ignore policy")
+					}
+				} else if input.Application.GroupLimit != nil {
+					if input.Application.GroupLimit.Policy == logging.DropPolicy {
+						el = append(el, AddThrottle(input)...)
+						inputs = append(inputs, fmt.Sprintf(UserDefinedSourceThrottle, input.Name))
+					} else {
+						log.V(1).Info("Not adding source in pipeline due to ignore policy")
+					}
+				} else {
+					inputs = append(inputs, fmt.Sprintf(UserDefinedAppInput, inputName))
+				}
```

Comment:
There are possible nil pointer errors here an some duplication  (double check my logic in case I missed somethign)
```suggestion
			if input, ok := userDefined[inputName]; ok && input.Application != nil {
			  // determine if a throttle is needed
				if (((app.ContainerLimit != nil && app.ContainerLimit.Policy != logging.IgnorePolicy) || 
				     (app.GroupLimit != nil && app.GroupLimit.Policy != logging.IgnorePolicy))) {
						el = append(el, AddThrottle(input)...)
						inputs = append(inputs, fmt.Sprintf(UserDefinedSourceThrottle, input.Name))
				} else {
					inputs = append(inputs, fmt.Sprintf(UserDefinedAppInput, inputName))
				}
```